Converted the classic pipeline to yaml Azure Templates CI

Harinder-accolite opened the pull request:
classic pipeline:-

repository: https://github.com/lansa/azure-quickstart-templates
branch: support/scalable
pipeline name - Template CI

converted classic to yaml pipeline

repository: Harinder-accolite/azure-quickstart-templates
branch: feature/harinder-classic2yaml
pipeline name: Azure Templates CI 
pipeline url: https://dev.azure.com/VisualLansa/Lansa%20Azure%20Scalable%20License%20Images/_build/results?buildId=2116&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a

Commit:
Created the pipeline need to create the service connection

Commit:
Made changes to remove predefined variables

Commit:
Configured Azure Logins variable group

Commit:
Commenting out

Commit:
created template ci 2.0 to 3.0 yaml

Commit:
Chianged the pipelines name to "Template-CI-3-$(Build.BuildId)" so that the template is validated.

Commit:
Changed the reference branch

Commit:
Changed the default value to Standard_v2

Commit:
made changes to task

Commit:
Moved the task to deleted an RG above the task that deploys ARM template.

Commit:
WIP

Commit:
wip

Commit:
Set continueOnError to true on the delete RG task.

Commit:
WIP

Commit:
WIP

Commit:
Removed system.debug variable and also set agpublicIpAllocationMethod to Dynamic.

Commit:
Accidentally pushed agpublicIpAllocationMethod as dynamic, while it should have been static. It's corrected now.

Commit:
WIP

Commit:
Removed the allowedValues  "Standard_Small",
"Standard_Medium",
"Standard_Large"

Commit:
Renamed the yaml file

Commit:
Changed the pipeline run-name from Template-CI-3.0-$(Build.BuildId) to Template-CI-$(Build.BuildId) and renamed the pipeline name to Azure Template CI.yaml

Commit:
Removed the resources section.

Commit:
Added the resources back

Commit:
Added task to delete the resource

Commit:
Added       ConnectedServiceName

Commit:
WIP

Commit:
WIP

Harinder-accolite commented:
Created the task to delete resource group as the Sql  Server (build-temsqlserver) was not getting deleted during the pipeline run.Is it okay that resource group is delete or we need to modify and delete only Sql Server.

robe070 commented:
> Created the task to delete resource group as the Sql Server (build-temsqlserver) was not getting deleted during the pipeline run.Is it okay that resource group is delete or we need to modify and delete only Sql Server.

Yes this is exactly what is required. Remove EVERYTHING that the creation of the Resource Group created. This is presuming that the resource group name is unique to this pipeline.

Commit:
Added the condition for resource group deletion

Harinder-accolite commented:
We have added the task to delete the resource group only when all the required task are successful.

robe070 reviewed (changes requested):
As you may not delete it, you also need to delete it if it exists BEFORE you deploy the template. This means that you cannot include the Build Number in the resource group name.

Change the name the Resource Group to 
```
BakingDP-Templates-CI-Test
```

And add a Task to delete the RG BEFORE deploying the template - and if it does not exist that is FINE. Ignore the error.

Commit:
Moved the task before the template deployment to check annd delete the resource group.

Commit:
Changed the resource group name to BakingDP-Templates-CI-Test

Harinder-accolite commented:
Moved the Task for task for RG deletion and replaced the RG name to "BakingDP-Templates-CI-Test" also added the condition to continue if no RG is available if a resource is available the task will delete the RG and then Deploy the Template.

robe070 commented on Azure Template CI.yaml:
```diff
@@ -328,10 +338,3 @@ jobs:
-      deploymentScope: 'Resource Group'
-      ConnectedServiceName: $(ConnectedServiceName)
-      action: 'DeleteRG'
-      resourceGroupName: BakingDP-TemplateTest-$(Build.BuildNumber)
```
I said to duplicate the task, not move it. You need to re-instate this task as it is.

robe070 replied:
You have still not done this!

robe070 commented on Azure Template CI.yaml:
```diff
@@ -77,13 +77,23 @@ jobs:
+      deploymentScope: 'Resource Group'
+      ConnectedServiceName: $(ConnectedServiceName)
+      action: 'DeleteRG'
+      resourceGroupName: BakingDP-Templates-CI-Test
```
You need to make the RG Name a YAML variable.

Harinder-accolite replied:
Created the RG Name as YAML variable.

robe070 reviewed (changes requested):
Hopefully you understand my comments this time.

Commit:
Task for checking and deletion of RG and continue on error.

robe070 reviewed (changes requested):
The changes you made are fine, you've left one out, as I've separately commented.

Commit:
Deletion of RG is added in end

Harinder-accolite committed:
Merge branch 'lansa:support/scalable' into feature/harinder-classic2yaml

Harinder-accolite commented:
Changes were made i.e task has been duplicated on the yaml and pushed to pipeline.

Commit:
Removed the condition which was to continue on error at end of template

Commit:
Merge branch 'feature/harinder-classic2yaml' of github.com:Harinder-accolite/azure-quickstart-templates into feature/harinder-classic2yaml

Harinder-accolite commented:
As per the discussion on the Call have deleted the task to continue on error at end of pipeline where as at the starting this task works fine.Ran the pipeline and artifact were created successfully .

robe070 commented:
Needed to change literal reference to resource group instead of $(ResourceGroup)

nkhl-accolite commented:
@robe070 - Do you mean you want to replace every "$(ResourceGroup)" with the actual value, which is "_BakingDP-Templates-CI-Test_"

robe070 commented:
What needs fixing? How will you know you've fixed it?

Leave it as it is. Run the pipeline. What's the symptom?
